test/Tests/src/Data/Text_Spec.enso

```diff
@@ -718,8 +718,7 @@ spec =
             'AB'.is_whitespace . should_be_false
             '  A   '.is_whitespace . should_be_false
 
-            # Disabled in the New Parser
-            # '\v\f\u{200a}\u{202f}\u{205F}\u{3000}\u{feff}'.is_whitespace . should_be_true
+            '\v\f\u{200a}\u{202f}\u{205F}\u{3000}'.is_whitespace . should_be_true
```

Comment:
This turned out to be a fun case. `U+FEFF`, the Unicode [Byte Order Mark](https://en.wikipedia.org/wiki/Byte_order_mark), is not actually whitespace (Java doesn't consider it whitespace: https://www.fileformat.info/info/unicode/char/feff/index.htm; apparently whatever implements `.trim` doesn't classify it as such either). The original test should never have passed; the reason it did is somehow the old parser *loses* BOM characters--even if they originate from explicit escape sequences! I assume they are being stripped implicitly in some conversion between string representations.

Since the new parser doesn't lose this character on the way to producing IR, now we have to fix these tests to be consistent with the behavior of `trim`.